subsys/bluetooth/host/adv.c

```diff
@@ -272,6 +272,25 @@ struct bt_le_ext_adv *bt_hci_adv_lookup_handle(uint8_t handle)
 #endif /* CONFIG_BT_BROADCASTER */
 #endif /* defined(CONFIG_BT_EXT_ADV) */
 
+struct bt_le_ext_adv *bt_adv_lookup_by_id(uint8_t id)
+{
+#if defined(CONFIG_BT_EXT_ADV)
+	for (size_t i = 0; i < ARRAY_SIZE(adv_pool); i++) {
+		if (atomic_test_bit(adv_pool[i].flags, BT_ADV_CREATED) &&
+		    adv_pool[i].id == id) {
+			return &adv_pool[i];
+		}
+	}
+#else
+	if (atomic_test_bit(bt_dev.adv.flags, BT_ADV_CREATED)) {
+		return &bt_dev.adv;
+	}
```

Comment:
Looks like this is missing `&& bt_dev.adv.id == id` to be symmetric with the other branch of the if-def. If I'm wrong, can you add an assert or code comment to explain why `id` is not used when `!CONFIG_BT_EXT_ADV`?